src/main/groovy/org/pih/warehouse/core/Historizable.groovy

```diff
@@ -0,0 +1,7 @@
+package org.pih.warehouse.core
+
+interface Historizable {
+    List<HistoryItem> getHistory()
+
+    ReferenceDocument getReferenceDocument()
```

Comment:
Interesting idea. However, I'm not sure I would put the reference document here since reference document is associated with the HistoryItem. In fact, each HistoryItem could have one or more ReferenceDocuments. For now we just might have the one, but the Received and Partially Received events might have a Receipt reference document, while Shipped might have a Shipment. I think the way it works out right now, Shipment is the reference document for each of these events but that's only the case because we don't have a Receipt details page that we want to show to the user.

Reply:
There is a ReferenceDocument in HistoryItem, so I see what you're doing here. Every class should decide how it should be rendered as a HistoryItem, including how we might render it as a link. So this method is just telling the class to provide an implementation for both levels (for the row and for a link within the row).

Reply:
Yeah, for now I assumed that a "workflow" - receipt, shipment would have only one type of reference document, especially that we do not have a receipt view page.